src/Language/PureScript/Bridge/SumType.hs

```diff
@@ -24,37 +26,60 @@ module Language.PureScript.Bridge.SumType (
 , recValue
 ) where
 
-import           Control.Lens      hiding (from, to)
+import           Control.Lens hiding (from, to)
+import           Data.List (nub)
+import           Data.Maybe (maybeToList)
 import           Data.Proxy
-import           Data.Set          (Set)
-import qualified Data.Set          as Set
-import           Data.Text         (Text)
-import qualified Data.Text         as T
+import           Data.Set (Set)
+import qualified Data.Set as Set
+import           Data.Text (Text)
+import qualified Data.Text as T
 import           Data.Typeable
 import           Generics.Deriving
 
 import           Language.PureScript.Bridge.TypeInfo
 
 -- | Generic representation of your Haskell types.
-data SumType (lang :: Language) = SumType (TypeInfo lang) [DataConstructor lang] deriving (Show, Eq)
+data SumType (lang :: Language) = SumType (TypeInfo lang) [DataConstructor lang] [Instance] deriving (Show, Eq)
 
 -- | TypInfo lens for 'SumType'.
 sumTypeInfo :: Functor f => (TypeInfo lang -> f (TypeInfo lang) ) -> SumType lang -> f (SumType lang)
-sumTypeInfo inj (SumType info constrs) = flip SumType constrs <$> inj info
+sumTypeInfo inj (SumType info constrs is) = (\ti -> SumType ti constrs is) <$> inj info
 
 -- | DataConstructor lens for 'SumType'.
 sumTypeConstructors :: Functor f => ([DataConstructor lang] -> f [DataConstructor lang]) -> SumType lang -> f (SumType lang)
-sumTypeConstructors inj (SumType info constrs) = SumType info <$> inj constrs
+sumTypeConstructors inj (SumType info constrs is) = (\cs -> SumType info cs is) <$> inj constrs
 
 -- | Create a representation of your sum (and product) types,
 --   for doing type translations and writing it out to your PureScript modules.
 --   In order to get the type information we use a dummy variable of type 'Proxy' (YourType).
 mkSumType :: forall t. (Generic t, Typeable t, GDataConstructor (Rep t))
           => Proxy t -> SumType 'Haskell
-mkSumType p = SumType  (mkTypeInfo p) constructors
+mkSumType p = SumType (mkTypeInfo p) constructors (Generic : maybeToList (nootype constructors))
   where
     constructors = gToConstructors (from (undefined :: t))
 
+-- | Purescript typeclass instances that can be generated for your Haskell types.
+data Instance = Generic | Newtype | Eq | Ord deriving (Eq, Show)
```

Comment:
Minor: Maybe rename those constructors to something like InstanceGeneric | InstanceNewType to avoid name clashes/confusion? What's your opinion on that?

Reply:
The possibility for confusion is definitely there. I drew inspiration (precedence, rather) from `aeson`, which has `String` as a valid constructor for the `Value` type. That, and the convenience of the `T.pack . show` combination you pointed out. 

Since `Instance` is "public" but not really a user concern (same as the actual structure of `SumType`), can we leave it the way it is?

Reply:
I believe so.